modules/tinymce/src/core/main/ts/content/InsertContent.ts

```diff
@@ -41,9 +42,13 @@ const processValue = (value: string | DetailsWithContent): ProcessedValue => {
 };
 
 const insertAtCaret = (editor: Editor, value: string | DetailsWithContent): void => {
-  const result = processValue(value);
+  const { content, details } = processValue(value);
 
-  Rtc.insertContent(editor, result.content, result.details);
+  preProcessSetContent(editor, { content, format: 'html', set: false, selection: true, paste: details.paste }).each((args) => {
+    Rtc.insertContent(editor, content, details);
+    postProcessSetContent(editor, content, args);
```

Comment:
This is incorrect... the `Rtc.insertContent` function needs to return the processed HTML and pass that to `postProcessSetContent` as that's what was passed previously here, not the input HTML.

Reply:
Ah oops. The plugin has never done that.

For set content RTC just re-serializes the HTML model returned by the TinyMCE parser, but that's still worth doing the same here rather than just using the input string. The parser will address things like security issues.

Reply:
Yeah, I just passed out the serialized input ast to get a value to send to the events setContent event I guess you should serialize the alt model fragment using the alt model serializer to get the lists and elements flattning like the actual inserted value. However the whole purpose from what I can see of the setContent event is to do things to the dom when contents gets inserted.

Not sure what the real need would be exactly to get the html being inserted after it has been inserted you can't really alter it at that point that is what the `beforeSetContent` event is for but all this was added before there where proper node filters etc. The events here makes less sense in the alt model. That is why I was not firing all of them.

If you look at every instance we use the `setContent` event it's not even caring about the event data it just does dom mutation or dom reading after some insert is made. So maybe we should deprecate that `content` property from the `setContent` event to me it serves little purpose. Also the serialization process is currently a bit expensive in the alt model since we unflatten marks, unflatten lists etc.

Reply:
No objections to deprecating it here. For RTC you could likely just return an empty string in that case.

Reply:
I added a ticket for that #TINY-8457 since those changes would hold up this one.